src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,81 @@
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import './TodoList.scss';
+import classNames from 'classnames';
+import { TodoShape } from '../../shapes/TodoShape';
+import { FilterField } from '../FilterField';
+import { TodoItem } from '../TodoItem';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends Component {
+  state = {
+    search: '',
+    shownTodos: 'all',
+  }
+
+  filterByAccomplishment = {
+    all: () => true,
+    active: todo => !todo.completed,
+    completed: todo => todo.completed,
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    const { todos, selectUser, shuffleTodos, selectedUserId } = this.props;
+    const { search, shownTodos } = this.state;
+    const { handleChange, filterByAccomplishment } = this;
+
+    const filteredTodos = todos
+      .filter(todo => todo.title.toLowerCase().includes(search.toLowerCase())
+        && filterByAccomplishment[shownTodos](todo));
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <FilterField
+          search={search}
+          handleChange={handleChange}
+          shownTodos={shownTodos}
+          shuffleTodos={shuffleTodos}
+        />
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {filteredTodos.map(({ id, userId, completed, title }) => (
+              <li
+                key={id}
+                className={classNames('TodoList__item',
+                  {
+                    'TodoList__item--unchecked': !completed,
+                    'TodoList__item--checked': completed,
+                  })}
```

Comment:
очень странно перенесена скобка
```suggestion
                className={classNames('TodoList__item',{
                    'TodoList__item--unchecked': !completed,
                    'TodoList__item--checked': completed,
                  })}
```